CRM/Civisocial/Backend/OAuthProvider/Facebook.php

```diff
@@ -0,0 +1,243 @@
+<?php
+require_once 'CRM/Civisocial/Backend/OAuthProvider.php';
+
+class CRM_Civisocial_Backend_OAuthProvider_Facebook extends CRM_Civisocial_Backend_OAuthProvider {
+
+	/**
+	 * Short name (alias) for OAuth provider
+	 *
+	 * @var string
+	 */
+	private $alias = 'facebook';
+
+	/**
+	 * Construct Facebook OAuth object
+	 *
+	 * @param string $accessToken
+	 *		Preobtained access token. Makes the OAuth Provider ready
+	 *		to make requests.
+	 */
+	public function __construct($accessToken = NULL) {
+		$this->apiUri = 'https://graph.facebook.com/v2.6';
+		$this->getApiCredentials($this->alias);
+		$this->token = $accessToken;
+	}
+
+	/**
+	 * Authorization URI that user will be redirected to for login
+	 *
+	 * @param array $permissions
+	 *		Permissions to be requested
+	 *
+	 * @return string | bool
+	 * @todo Check if requests have been reviewed by Facebook
+	 */
+	public function getLoginUri($permissions = array(), $reRequest = FALSE) {
+		$uri = 'https://www.facebook.com/dialog/oauth';
+		$params = array(
+			'client_id' => $this->apiKey,
+			'redirect_uri' => $this->getCallbackUri($this->alias),
+		);
+		if (empty($permissions)) {
+			$params['scope'] = implode(',', array_merge($this->getBasicPermissions(), $this->getExtraPermissions()));
+		} else {
+			$params['scope'] = implode(',', $permissions);
```

Comment:
This function takes an array of permissions, so in future a developer might want to use this with some specific permission, in that case having basic permission appended would help.